src/main/java/uk/ac/ebi/ega/egafuse/validator/Validator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.validator;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import jnr.posix.util.Platform;
+import joptsimple.OptionSet;
+import joptsimple.internal.Strings;
+
+public class Validator {
+    public static boolean isValid(OptionSet optionSet) throws IOException {
+        if (!Platform.IS_WINDOWS) {
+            File mntTest = new File(optionSet.valueOf("m").toString());
+            if (!mntTest.exists() && !mntTest.isDirectory()) {
+                throw new IllegalArgumentException(mntTest.getPath()
+                        .concat(" can't be used as mount point. Ensure that the directory path exists and is empty"));
+            }
+        }
+
+        if (optionSet.has("cf")) {
+            String credFilePath = optionSet.valueOf("cf").toString();
+            Map<String, String> credentials = readFileArguments(credFilePath);
+            if (Strings.isNullOrEmpty(credentials.get("username"))
+                    || Strings.isNullOrEmpty(credentials.get("password"))) {
+                throw new IllegalArgumentException("Username or Password not Specified in File ".concat(credFilePath));
+            }
+        }
+
+        return true;
+    }
+
+    private static Map<String, String> readFileArguments(String filepath) throws IOException {
```

Comment:
If this function is only suppossed to return the user / password / grant then it is better if you return an object with those members instead of a generic map. And if while parsing you can't find the parameters throw the error here. Instead of rechecking the values again. In fact I would move this into the parser and make the parser class return an object with all the informations tha is needed instead of then later needing to call this method again to get the data again.

Reply:
moved into parser class to validate and return username and password